libgalois/include/katana/analytics/Plan.h

```diff
@@ -28,10 +28,13 @@ enum Architecture {
 /// efficient. This may be useful, for instance, to use a sampled subgraph to compute a plan for use on a the original
 /// larger graph.
 class Plan {
-protected:
   Architecture architecture_;
 
-  Plan(Architecture architecture) : architecture_(architecture) {}
+protected:
+  explicit Plan(Architecture architecture) : architecture_(architecture) {}
+  void SetArchitecture(const Architecture architecture) {
+    architecture_ = architecture;
+  }
```

Comment:
Do we actually need this setter? I would love to not have it at all. Subclasses can just read the value and not set it and immutability is good.

Reply:
We need it because we need to modify the plan, as per your request.

Reply:
I'm also generally severely allergic to `Set` methods. Would it make sense to replace set calls with `myplan = Plan(NewArchitecture)`?

Reply:
No, it is not possible because the setter is used during the construction of the derived class. Here is an example: https://github.com/KatanaGraph/katana-enterprise/blob/ee91de57d6654ad74d0b14b30b81bdf4a696143a/libdistgalois/src/dist-analytics/sssp/Sssp.cpp#L161
We do not have a `plan` variable to modify - we have to replace the value of the field in some cases.